webapp-django/crashstats/base/tests/test_ga.py

```diff
@@ -14,16 +16,30 @@
 from crashstats.base.ga import track_api_pageview, track_pageview
 
 
+EXPECTED_CID = 'ab8f0910428745d9995da0c51a9d3b64'
+
+
+def mock_uuid4(fun):
+    """Mock uuid.uuid4() decorator that returns a real non-random UUID"""
+    @wraps(fun)
+    def _mock_uuid4(*args, **kwargs):
+        with mock.patch('uuid.uuid4') as mocked_uuid:
+            mocked_uuid.return_value = UUID(hex=EXPECTED_CID, version=4)
+            return fun(*args, **kwargs)
+
+    return _mock_uuid4
```

Comment:
This is sort of like what I'm doing with Antenna. The thinking is that mocking decorators and context managers provide a better testing infrastructure API than off-the-cuff mocking.

But if you don't like it, it's entirely in the last commit and we can nix that and go back to mocking uuid in each test that uses it.